src/hotspot/share/c1/c1_Instruction.hpp

```diff
@@ -334,7 +334,7 @@ class Instruction: public CompilationResourceObj {
   };
 
  public:
-  void* operator new(size_t size) throw() {
+  void* operator new(size_t size) noexcept {
```

Comment:
Should the noexcept specifiers on both operator new in this file be removed? They both call Amalloc with the default parameter of EXIT\_OOM, which means they both crash instead of returning nullptr when failing to allocate memory

Reply:
I think EXIT_OOM is an implementation detail. The key point is that these methods do not throw.

Reply:
Or do I have that backwards ... the key point of `no_except` is that callers of these method must check for `nullptr`, but with EXIT_OOM that is not the case - and we don't want it to appear that the allocation can actually fail and we continue execution!

Reply:
It looks like there are a number of `operator new`s that have nothrow exception specs but shouldn't.
CompilationResourceObj in the immediately preceding file, for example.
A precursor cleanup (or maybe several) that removed those first would be nice.

Reply:
It seems that https://bugs.openjdk.org/browse/JDK-8305590 only removed some of them.